marlowe-dashboard-client/src/Types.purs

```diff
@@ -1,27 +1,33 @@
 module Types
   ( WebData
+  , DecodedWebData
   , ContractInstanceId(..)
-  , contractInstanceIdFromString
   , MarloweParams
   , ValidatorHash
+  , MarloweData
   ) where
 
 -- this module is for miscellaneous types that don't belong anywhere else
 -- (or would lead to cyclic dependencies if put somewhere they might more naturall belong)
 import Prelude
+import Data.Either (Either)
 import Data.Generic.Rep (class Generic)
-import Data.Maybe (Maybe(..))
+import Data.List.NonEmpty (NonEmptyList)
 import Data.Newtype (class Newtype)
-import Data.UUID (UUID, parseUUID)
+import Data.UUID (UUID)
+import Foreign (ForeignError)
 import Foreign.Class (class Encode, class Decode)
 import Foreign.Generic (defaultOptions, genericDecode, genericEncode)
-import Marlowe.Semantics (CurrencySymbol)
+import Marlowe.Semantics (Contract, CurrencySymbol, State)
 import Network.RemoteData (RemoteData)
 import Servant.PureScript.Ajax (AjaxError)
 
 type WebData
   = RemoteData AjaxError
 
+type DecodedWebData
+  = RemoteData (Either AjaxError (NonEmptyList ForeignError))
```

Comment:
If later on, we need to add yet another error, this might be handy:
https://github.com/natefaubion/purescript-checked-exceptions